### PR TITLE
implemented most of the functionality described in the task

### PR DESCRIPTION
**Pull request by hobbitJS**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,29 +1,17 @@
 <template>
-  <div id="app">
-    <div id="nav">
-      <router-link to="/">Home</router-link> |
-      <router-link to="/issues">Issues</router-link>
-    </div>
-    <router-view/>
+  <div>
+    <Navbar/>
+    <router-view></router-view>
```
Comment (gregurco) [inline src/App.vue]: you can write `<router-view />`

File: src/components/Comment.vue [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<template>
+  <div>
+    <hr>
+    <div>
+      <img
+        :src="comment.user.avatar_url"
+        class="rounded float-left"
+        style="width: 75px; height: 75px; marginRight: 20px"
```
Comment (gregurco) [inline src/components/Comment.vue]: Avoid inline styles. Better use scoped styles.

File: src/components/Comment.vue [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        alt="..."
+      >
+      <div class="card">
+        <div class="card-header">
+          <span style="fontWeight: bold">{{comment.user.login}}</span>
```
Comment (gregurco) [inline src/components/Comment.vue]: code style: `{{ comment.user.login }}`
and also inline styles

File: src/components/Issue.vue [hunk 4]
```diff
@@ -1,24 +1,35 @@
...
-        </div>
-        <div class="card-body">
-          <p class="card-text">#{{issue.number}} opened {{issue.created_at}} by {{issue.user.login}}</p>
-        </div>
+  <div class="card issue">
+    <div class="card-body">
+      <div style="position: relative">
+        <span
+          class="badge"
+          :class="issue.state === 'open' || issue.state === 'all' ? 'badge-success' : 'badge-danger'"
```
Comment (gregurco) [inline src/components/Issue.vue]: quite complicated to understand this class definition. Use next mode: `:class="{ classNameA: condition1: classNameB: condition2 }"`
More here: https://vuejs.org/v2/guide/class-and-style.html#Object-Syntax

File: src/components/IssueDetail.vue [hunk 5]
```diff
@@ -1,36 +1,116 @@
...
-      issue: 'GET_ISSUE_BY_ID'
-    })
-  },
-  mounted () {
-    this.set(this.$route.params.id)
+      issue: 'GET_SELECTED_ISSUE',
+      isLoading: 'GET_ISLOADING',
+      comments: 'GET_COMMENTS'
+    }),
+    date () {
```
Comment (gregurco) [inline src/components/IssueDetail.vue]: you already have the same behavior in another component. This about implementing of mixins or filters. More info here:
- https://vuejs.org/v2/guide/mixins.html
- https://vuejs.org/v2/guide/filters.html

File: src/components/IssueDetail.vue [hunk 6]
```diff
@@ -1,36 +1,116 @@
...
+      setIsLoading: 'SET_ISLOADING',
+      setComments: 'SET_COMMENTS'
     })
+  },
+  watch: {
+    issue (newValue, oldValue) {
+      const newNumber = newValue !== null ? newValue.number : null
+      const oldNumber = oldValue !== null ? oldValue.number : null
+
+      if (newNumber !== oldNumber) {
```
Comment (gregurco) [inline src/components/IssueDetail.vue:79]: watch is triggered when value is changes. Is it possible that old value will be the same as new one?

File: src/components/IssueStateSetter.vue [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  <div class="d-flex flex-row card" style="padding: 5px 5px">
+    <div class="card-header">Filter</div>
+    <div class="btn-group float-right" role="group">
+      <button
+        v-for="button in buttons"
+        type="button"
+        class="btn btn-secondary"
+        @click="setState(button.toLowerCase())"
+        :key="button"
+        :class=" itemsState === button.toLowerCase() ? 'active' : null"
```
Comment (gregurco) [inline src/components/IssueStateSetter.vue]: also better use class definition in object style

File: src/components/IssueStateSetter.vue [hunk 8]
```diff
@@ -0,0 +1,41 @@
+<template>
+  <div class="d-flex flex-row card" style="padding: 5px 5px">
+    <div class="card-header">Filter</div>
+    <div class="btn-group float-right" role="group">
+      <button
+        v-for="button in buttons"
+        type="button"
+        class="btn btn-secondary"
+        @click="setState(button.toLowerCase())"
```
Comment (gregurco) [inline src/components/IssueStateSetter.vue:9]: why not to call `setItemsState` directly?

File: src/components/Navbar.vue [hunk 9]
```diff
@@ -0,0 +1,43 @@
+<template>
+  <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
+    <router-link class="navbar-brand" to="/">Github</router-link>
```
Comment (gregurco) [inline src/components/Navbar.vue]: better: `:to="{ name: 'routeName' }"`

File: src/store.js [hunk 10]
```diff
@@ -11,33 +11,72 @@ let axiosClient = axios.create({
 export default new Vuex.Store({
   state: {
     list: [],
-    selectedIssue: null
+    selectedIssue: null,
+    comments: [],
+    firstPage: 1,
```
Comment (gregurco) [inline src/store.js]: why this property is called `firstPage`? What if you setup there 2 or 3 selecting another page?

File: src/App.vue (hunk 1 above)
Comment (hobbitJS) [reply src/App.vue]: Fixed.

File: src/components/Comment.vue (hunk 2 above)
Comment (hobbitJS) [reply src/components/Comment.vue]: Created a class "avatar-img".

File: src/components/Comment.vue (hunk 3 above)
Comment (hobbitJS) [reply src/components/Comment.vue]: fixed code style. 
added bootstrap class "font-weight-bold".

File: src/components/Issue.vue (hunk 4 above)
Comment (hobbitJS) [reply src/components/Issue.vue]: Created a method called "detectState(state)" which returns the convenient class depending on "issue.state"

File: src/components/IssueDetail.vue (hunk 5 above)
Comment (hobbitJS) [reply src/components/IssueDetail.vue]: tried to create a mxin and use it, but the app was crashed , could you purpose a way how to solve?

File: src/components/IssueStateSetter.vue (hunk 7 above)
Comment (hobbitJS) [reply src/components/IssueStateSetter.vue]: created method "detectState" which return the convinient class depending on state.

File: src/components/IssueStateSetter.vue (hunk 8 above)
Comment (hobbitJS) [reply src/components/IssueStateSetter.vue:9]: Fixed.

File: src/components/Navbar.vue (hunk 9 above)
Comment (hobbitJS) [reply src/components/Navbar.vue]: Fixed.

File: src/store.js (hunk 10 above)
Comment (hobbitJS) [reply src/store.js]: Changed the name from "firstPage" to "firstPaginationButton".

File: src/components/IssueDetail.vue (hunk 5 above)
Comment (gregurco) [reply src/components/IssueDetail.vue]: It depends on the error. I used mixins and it was ok.

File: src/components/Issue.vue [hunk 11]
```diff
@@ -1,24 +1,49 @@
...
-          <p class="card-text">#{{issue.number}} opened {{issue.created_at}} by {{issue.user.login}}</p>
-        </div>
+  <div class="card issue">
+    <div class="card-body">
+      <div class="position-relative">
+        <span
+          class="badge issue-state-badge"
+          :class="detectState(issue.state)"
+        >!</span>
+        <router-link :to="{name: 'IssueDetail', params: { number: issue.number }}">
```
Comment (gregurco) [inline src/components/Issue.vue]: Code style here. Use lint.

File: src/store.js [hunk 12]
```diff
@@ -11,33 +11,72 @@ let axiosClient = axios.create({
 export default new Vuex.Store({
   state: {
     list: [],
-    selectedIssue: null
+    selectedIssue: null,
+    comments: [],
+    firstPaginationButton: 1,
```
Comment (gregurco) [inline src/store.js]: call it `page`

Comment (hobbitJS) [reply src/store.js]: fixed.

File: src/components/Issue.vue (hunk 11 above)
Comment (hobbitJS) [reply src/components/Issue.vue]: fixed

File: src/components/IssueDetail.vue (hunk 5 above)
Comment (hobbitJS) [reply src/components/IssueDetail.vue]: created a folder called filters, with 2 files (detectState, formatData)

**Comment by gregurco** [comment]
:+1:

**Pull request merged**